java/algorithm/src/main/java/org/apache/arrow/algorithm/sort/DefaultVectorComparators.java

```diff
@@ -115,6 +120,11 @@ public int compareNotNull(int index1, int index2) {
       short value2 = vector2.get(index2);
       return value1 - value2;
     }
+
+    @Override
+    public VectorValueComparator<SmallIntVector> clone() {
+      return new ShortComparator();
```

Comment:
if this is creating brnd new comparators without the attached vectors, then clone/copy is probably the wrong name.  createNew or something similar is probably more appropriate.

Reply:
createNew sounds good to me. Revised accordingly.